lib/puppet/file_bucket/dipper.rb

```diff
@@ -22,7 +22,7 @@ def initialize(hash = {})
       @rest_path  = nil
     else
       @local_path = nil
-      @rest_path = "https://#{server}:#{port}/#{environment}/file_bucket_file/"
+      @rest_path = ""
```

Comment:
Hi @ffrank, this "works" because the puppet will use the local filebucket instead of the remote one.

With this line changed, when I backup an `empty` file, the request object is:

```
$ bundle exec puppet filebucket backup empty --server hwvft1j1a4531gm.delivery.puppetlabs.net --http_debug
...
(rdb:1) request
#<Puppet::Indirector::Request:0x007fa919a6e770 @instance=nil, @indirection_name=:file_bucket_file, @method=:head, @options={}, @key="md5/d41d8cd98f00b204e9800998ecf8427e//Users/josh/work/puppet/empty">
(rdb:2) request.protocol
nil
```

The default terminus for the file_bucket is the [`:selector`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/indirector/file_bucket_file/selector.rb#L7-L13). When the `request.protocol` is nil it selects the `:file` terminus (the local one).

After puppet runs, you'll see the backed up file in the local filebucket, even though I've specified a remote server.

```
$ find /Users/josh/.puppetlabs/opt/puppet/cache/bucket/ -name paths -type f -exec cat {} \; -print
/Users/josh/work/puppet/empty
/Users/josh/.puppetlabs/opt/puppet/cache/bucket//d/4/1/d/8/c/d/9/d41d8cd98f00b204e9800998ecf8427e/paths
```